smallrye-reactive-messaging-kafka-api/src/main/java/io/smallrye/reactive/messaging/kafka/api/IncomingKafkaRecordBatchMetadata.java

```diff
@@ -0,0 +1,46 @@
+package io.smallrye.reactive.messaging.kafka.api;
+
+import java.util.Set;
+
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.common.TopicPartition;
+
+/**
+ * Contains information about the batch of messages received from a channel backed by Kafka.
+ * Encapsulates underlying Kafka {@link ConsumerRecords} received from the consumer client.
+ *
+ * As this is an incoming message metadata it is created by the framework and injected into incoming batch messages.
+ *
+ * @param <K> The record key type
+ * @param <T> The record payload type
+ */
+public class IncomingKafkaRecordBatchMetadata<K, T> {
+
+    private final ConsumerRecords<K, T> records;
+
+    public IncomingKafkaRecordBatchMetadata(ConsumerRecords<K, T> records) {
+        this.records = records;
+    }
+
+    /**
+     * @return the underlying Kafka {@link ConsumerRecords}
+     */
+    public ConsumerRecords<K, T> getRecords() {
+        return records;
+    }
+
+    /**
+     * @return the number of records for all topics
```

Comment:
"for all topics" -> "for all partitions"? Or just "the total number of records in this record batch"?

Reply:
I think the term "topic partitions" is the most appropriate one. I've shamelessly copied the description for `ConsumerRecords#count`.